lib/deaf_grandma.rb

```diff
@@ -17,11 +16,24 @@ def run!
     end
   end
 
-
   def speak(input)
+    case input
+    when -> (i) { i == 'BYE' } then @bye_counter += 1
+    when -> (i) { i == i.upcase } then return 'NOT SINCE 1964!'
+    else return 'SPEAK UP SONNY!'
+    end
 
-    #Implement your code here <<<<<<<<<
+    if @bye_counter == 3
+      p 'SEE YOU LATER SONNY!'
+      exit
+    end
 
+    'NOT SINCE 1964!'
```

Comment:
It hits this if you say 'BYE' and nothing else. I could have made it more concise, I agree.